### PR TITLE
Alphanumeric New Relic licence keys

### PR DESCRIPTION
**Pull request by johnsyweb**
Context
----------

Our valid, New Relic-provided licence key contains 40 alphanumeric characters, but is rejected by the `nrsysmond` module because some of the characters are not hexadecimal.

Changes
----------
- Fix `Vagrantfile`, in order to test
- Pull in #7, in order to test (thanks @jantman, you saved me some time)
- Add specs around validating `license_key` (too long, invalid characters)
- Tighten regular expression to catch keys being too long
- Add spec to highlight our issue
- Allow all (ASCII) alphanumeric characters

Cc
---
@jessedearing

### CODE REVIEW

**Comment by denmat** [comment]
Good work @johnsyweb

File: Vagrantfile [hunk 1]
```diff
@@ -7,12 +7,12 @@ Vagrant::Config.run do |config|
   # please see the online documentation at vagrantup.com.
 
   # Every Vagrant virtual environment requires a box to build off of.
-  config.vm.box = "centos-6.3-64bit-puppet-vbox.4.1.20-2"
+  config.vm.box = 'CentOS-6.4-x86_64-v20131103.box'
```
Comment (andrewjhumphrey) [inline Vagrantfile]: Why are we using centos as the default?  Given the vast bulk of our machines are debian/ubuntu...

**Comment by andrewjhumphrey** [comment]
Niggle aside.  Nice work!

**Comment by ejoubaud** [comment]
Lol, I was just looking into forking it myself (which apparently I can't do to the Envato account anyway :'( ), when I saw your face on their fork graph :)

Awesome work :100:

**Comment by jessedearing** [comment]
LGTM :+1:

**Pull request merged**